src/lug/Window/Window.cpp

```diff
@@ -0,0 +1,70 @@
+#include <lug/Window/Window.hpp>
+
+#if defined(LUG_SYSTEM_WINDOWS)
+
+    #include <lug/Window/Win32/WindowImplWin32.hpp>
+
+#elif defined(LUG_SYSTEM_LINUX)
+
+    #include <lug/Window/Unix/WindowImplX11.hpp>
+
+#elif defined(LUG_SYSTEM_ANDROID)
+
+    #include <lug/Window/Android/WindowImplAndroid.hpp>
+
+#endif
+
+lug::Window::Window::Window() : _impl(new lug::Window::priv::WindowImpl()) {
+}
+
+lug::Window::Window::~Window() {
+    close();
+    delete _impl;
+}
+
+std::unique_ptr<lug::Window::Window> lug::Window::Window::create(uint16_t width, uint16_t height, const std::string& title, Style style) {
+    std::unique_ptr<Window> win(new Window());
+
+    if (!win->createWindow(width, height, title, style)) {
+        return nullptr;
+    }
+
+    return std::move(win);
+}
+
+bool lug::Window::Window::createWindow(uint16_t width, uint16_t height, const std::string& title, Style style) {
+    if (_impl != nullptr) {
+        return _impl->create(width, height, title, style);
+    }
+
+    return false;
+}
+
+bool lug::Window::Window::isOpen( ) const {
+    return _impl != nullptr ? true : false;
+}
+
+bool lug::Window::Window::pollEvent(Event& event) {
+    if (_events.empty() && _impl != nullptr) {
+        _impl->processEvents(_events);
+    }
+
+    if (!_events.empty()) {
+        event = _events.front();
+        _events.pop();
+
+        return true;
+    }
+
+    return false;
+}
+
+bool lug::Window::Window::close() {
```

Comment:
Does a `close` can fail? If not, do not return `bool`
I think that it doesn't matter if `_impl` is `nullptr` because the window is already closed.